develop/tutorials/articles/03-portlets/packaging-a-jsf-application.markdown

```diff
@@ -72,19 +113,13 @@ that extends
 - The `bnd.bnd` file can be processed by a build-time plugin (e.g.,
   [bnd-maven-plugin](http://njbartlett.name/2015/03/27/announcing-bnd-maven-plugin.html))
   to affect the content of an OSGi-ready `META-INF/MANIFEST.MF`.
-- Can take advantage of OSGi Declarative Services (DS).
-- Provides a class that extends
-  [GenericFacesPortlet](http://myfaces.apache.org/portlet-bridge/2.0/api/apidocs/javax/portlet/faces/GenericFacesPortlet.html)
-  and specifies portlet attributes via the `@Component` annotation, rather than
-  using two separate descriptors: `WEB-INF/portlet.xml` and
-  `WEB-INF/liferay-portlet.xml`.
 
 **Drawbacks:**
 
-- Bypasses the Liferay auto-deploy process, which means developers must have
-  the `WEB-INF/web.xml` descriptor fully ready for deployment, such as
-  `<servlet>` and `<servlet-mapping>` for Liferay's `PortletServlet` and a
+- Bypasses the Liferay auto-deploy process, and the WAB Generator.
+- Developers must have the `WEB-INF/web.xml` descriptor fully ready for deployment with a
```

Comment:
@vsingleton I *think* it also needs to specify a ```<servlet>``` and ```<servlet-mapping>``` for the PortletServlet. It also needs to include the ```init-param``` and the class that extends GenericFacesPortlet needs to have the ```setServlet(Servlet)``` method in order to deal with the thread-timing problem with DS WABs.